graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/MutationFetcher.java

```diff
@@ -74,9 +77,33 @@ protected MutationFetcher(
     }
 
     // Execute as a single statement
-    return dataStore
-        .execute(query)
-        .thenApply(rs -> ImmutableMap.of("value", environment.getArgument("value")));
+    return dataStore.execute(query).thenApply(rs -> toMutationResult(rs, environment));
+  }
+
+  private Map<String, Object> toMutationResult(
+      ResultSet resultSet, DataFetchingEnvironment environment) {
+
+    List<Row> rows = resultSet.currentPageRows();
+    Object originalValue = environment.getArgument("value");
+
+    if (rows.isEmpty()) {
+      // if we have no rows means that we got no information back from query execution, return
+      // original value to the user
+      return ImmutableMap.of("value", originalValue);
```

Comment:
Is it actually meaningful to the user? The user already has the original value :thinking:

Reply:
Exactly what I asked in the PR description :thinking:

Reply:
Sorry, TL;DR ;-)

Reply:
Would we also return `applied` is `true` here?

Reply:
@mpenick Good question, I also have code it at first like this.. 

However, I was not sure if we can surely say that it was applied here? Reading Cassandra CQL reference I was a bit confused, not sure what happens if you hit insert with same primary keys twice?

Reply:
> Is it actually meaningful to the user? The user already has the original value 🤔

Legitimate question, but since the existing code was already doing it I think we have to preserve backward compatibility.

> Should we also return `applied` is `true` here?

I would lean towards yes. `applied` is only relevant for lightweight transactions, but you could say that if a query is not a LWT, then it's not a failed LWT, therefore it was "applied" in the sense that the backend did not reject it (regardless of whether it actually changed any data). That's how we did it in the Java driver at least.

Reply:
Although adding `applied` to the response to ordinary mutations makes sense to me, I think that will invalidate our backwards compatibility since we cannot mix it into the existing (original) value that is currently returned in this case. WDYT?

Reply:
If it's a completely new field so it won't break compatibility. Legacy requests did not include it in the selection set, so they'll keep returning exactly the same response.

Reply:
Since this is also how it's done in the Java driver, maybe we can do it then.. It would seem a bit odd for the delete though, you would say applied, although the delete might not remove any data.. But if we look it from the *the sense that the backend did not reject it*, then it's fine I guess..